providers/amazon/tests/provider_tests/amazon/aws/links/test_base_aws.py

```diff
@@ -194,14 +194,10 @@ def assert_extra_link_url(
         )
 
         error_msg = f"{self.full_qualname!r} should be preserved after execution"
-        assert ti.task.get_extra_links(ti, self.link_class.name) == expected_url, error_msg
 
-        serialized_dag = self.dag_maker.get_serialized_data()
-        deserialized_dag = SerializedDAG.from_dict(serialized_dag)
-        deserialized_task = deserialized_dag.task_dict[self.task_id]
-
-        error_msg = f"{self.full_qualname!r} should be preserved in deserialized tasks after execution"
-        assert deserialized_task.get_extra_links(ti, self.link_class.name) == expected_url, error_msg
+        # This is the right thing to test in the provider as providers shouldn't influence ti.task
+        # This is also the source of the link, if all's fine here, the link will be pushed to xcom and fetched too
```

Comment:
```suggestion
```

This is correct, but to me doesn't serve value as a long term code comment.

Reply:
Yeah sure removing